### ISSUE TITLE
[no-misused-new] False positive when used on an interface for constructor function which constructs itself

### ISSUE THREAD

**Issue by miklcct** [typescript-eslint/typescript-eslint#2697]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "parser" : "@typescript-eslint/parser",
    "plugins" : [
        "@typescript-eslint"
    ],
    "rules": {
        "@typescript-eslint/no-misused-new": 2
    }
}
```

```TS
// your repro code case
interface SelfReproducingClass {
    new(): SelfReproducingClass
}
```

**Expected Result**

I expect no error.

**Actual Result**

The rule produces an error.

**Additional Info**

A constructor function satisfying `SelfReproducingClass` is a constructor function that, when called as `new SelfReproducingClass`, will return a constructor function which also satisfies the same interface. [Here's an implementation in Javascript](https://github.com/miklcct/js-self-reproducing-class/blob/5b64b7c0f9a213995363e17546e4bad6781b8395/index.js).

The error can be worked around by adding `& unknown` into the type declaration:
```TS
interface SelfReproducingClass {
    new(): SelfReproducingClass & unknown
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.5.0` |
| `@typescript-eslint/parser`        | `4.5.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.11.0` |
| `node`                             | `14.13.0` |

**Comment by bradzacher** [comment]
This is one of the very few cases the rule intentionally reports against.
If you are intentionally using this pattern for some reason, then you should use a disable comment or disable the rule.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-new.md
